sdk-lib/src/androidTest/java/cash/z/ecc/android/sdk/internal/storage/block/FileCompactBlockRepositoryTest.kt

```diff
@@ -0,0 +1,131 @@
+        }
+        if (blocksDir.existsSuspend()) {
+            blocksDir.deleteRecursivelySuspend()
+        }
+        blocksDir.mkdirsSuspend()
+
+        compactBlockRepository =
+            FileCompactBlockRepository(rootDir, rustBackend)
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @After
+    fun tearDown() = runTest {
+        File(DatabasePathFixture.new()).deleteRecursivelySuspend()
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Test
+    fun getLatestHeightTest() = runTest {
+        val blockRange = FileBlockRangeFixture.new()
+
+        val lightwalletService = LightWalletGrpcService.new(context, lightwalletdHost)
+        val blocks = lightwalletService.getBlockRange(blockRange).toList()
+
+        compactBlockRepository.write(blocks.asSequence())
+
+        assertTrue { blocks.last().height == compactBlockRepository.getLatestHeight()?.value }
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Test
+    fun writeBlockTest() = runTest {
+        val blockRange = FileBlockRangeFixture.new()
+
+        val lightwalletService = LightWalletGrpcService.new(context, lightwalletdHost)
+        val blocks = lightwalletService.getBlockRange(blockRange)
+
+        assertTrue { rustBackend.metadata.size == 0 }
+
+        compactBlockRepository.write(blocks)
+        assertTrue { rustBackend.metadata.size == 10 }
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Test
+    fun rewindToTest() = runTest {
+        val blockRange = FileBlockRangeFixture.new()
+
+        val lightwalletService = LightWalletGrpcService.new(context, lightwalletdHost)
+        val blocks = lightwalletService.getBlockRange(blockRange)
+
+        val rewindHeight: Long = 500_005
+        compactBlockRepository.write(blocks)
+
+        compactBlockRepository.rewindTo(BlockHeight(rewindHeight))
+
+        val metaData = rustBackend.metadata.filter { it.height > rewindHeight }
+        assertTrue { metaData.isEmpty() }
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Test
+    fun createTemporaryFileTest() = runTest {
+        val blockRange = FileBlockRangeFixture.new()
+
+        val lightwalletService = LightWalletGrpcService.new(context, lightwalletdHost)
+        val blocks = lightwalletService.getBlockRange(blockRange)
+        val block = blocks.first()
+
+        val file = block.createTemporaryFile(FilePathFixture.newBlocksDir())
+
+        assertTrue { file.existsSuspend() }
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Test
+    fun finalizeFileTest() = runTest {
+        val blockRange = FileBlockRangeFixture.new()
+
+        val lightwalletService = LightWalletGrpcService.new(context, lightwalletdHost)
+        val blocks = lightwalletService.getBlockRange(blockRange)
+        val block = blocks.first()
+
+        val file = block.createTemporaryFile(FilePathFixture.newBlocksDir())
+        val finalizedFile = File(file.absolutePath.dropLast(4))
```

Comment:
We can make this more robust by using the length of the file extension, which we have as a constant already.